lib/internal/Magento/Framework/App/Http.php

```diff
@@ -153,6 +153,8 @@ public function launch()
      */
     public function catchException(Bootstrap $bootstrap, \Exception $exception)
     {
+        $this->_eventManager->dispatch('application_handled_exception', ['exception' => $exception]);
```

Comment:
New events must not be created, you can implement this logic as a before plugin for `catchException` method.

Reply:
@orlangur i refactored it to use a plugin.